src/Microsoft.Health.Fhir.Core/Features/Context/ScopeRestriction.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Health.Fhir.Core.Features.Security;
+
+namespace Microsoft.Health.Fhir.Core.Features.Context
+{
+    public class ScopeRestriction : IEquatable<ScopeRestriction>
+    {
+        public ScopeRestriction(string resource, DataActions allowedAction, string user)
+        {
+            Resource = resource;
+            AllowedDataAction |= allowedAction;
+            User = user;
+        }
```

Comment:
Are null values allowed?

Reply:
I added a check for null for Resource, but we may allow nulls for user.